lms/djangoapps/student_dashboard/test/test_views.py

```diff
@@ -0,0 +1,111 @@
+        enrollment_end=datetime.utcnow() + timedelta(days=2),
+    )
+    interests = ['interest_strategy_planning', 'interest_leadership_governance', 'interest_program_design']
+
+    @factory.django.mute_signals(signals.pre_save, signals.post_save)
+    def setUp(self):
+        self.user = UserFactory()
+        self.request = RequestFactory()
+        self.request.user = self.user
+        self.course = CourseFactory.create()
+        self.course_enrollments = [
+            CourseEnrollmentFactory.create(user=self.user, course_id=self.course.id, is_active=True,
+                                           course__start=datetime.utcnow() - timedelta(days=2),
+                                           course__end=datetime.utcnow() + timedelta(days=2))]
+        self.course_card = CourseCardFactory(course_id=self.course.id, course_name=self.course.name)
+        self.course_overview = CourseOverviewFactory(display_name='test_overview',
+                                                     **TestStudenDashboard.course_overview_kwargs)
+
+        self.orphan_course = CourseFactory.create()
+        self.orphan_course_overview = CourseOverviewFactory(display_name='orphan_test_overview',
+                                                            id=self.orphan_course.id,
+                                                            **TestStudenDashboard.course_overview_kwargs)
+        self.orphan_course_card = CourseCardFactory(course_id=self.orphan_course.id,
+                                                    course_name=self.orphan_course.name)
+
+        self._init()
+
+    def _init(self):
+        """
+        _init method is for data set initialization.
+        """
+        self.user.extended_profile.interests = TestStudenDashboard.interests
+        self.user.extended_profile.save()
+
+        _settings, _created = CustomSettings.objects.get_or_create(id=self.course.id)
+        _orphan_settings, _orphan_created = CustomSettings.objects.get_or_create(id=self.orphan_course.id)
+
+        _settings.tags = \
+            _orphan_settings.tags = 'Strategy and planning|Leadership and governance|Program design and development'
+
+        _settings.save()
+        _orphan_settings.save()
+
+        initialize_course_settings(self.course, self.course_overview, True)
+        initialize_course_settings(self.orphan_course, self.orphan_course_overview, True)
+
+    @ddt.data(200, 500)
+    def test_get_joined_communities(self, status):
+        """
+        test_get_joined_communities test method is covering
+        lms.djangoapps.student_dashboard.views.get_joined_communities
+        """
+        with mock.patch.object(ForumCategory, 'joined') as joined:
+            joined.return_value = status, []
+            communities = get_joined_communities('some-username')
+            self.assertEqual(communities, [])
+
+    @ddt.data('onboarding', 'other')
+    def test_get_recommended_xmodule_courses(self, _from):
```

Comment:
Great work on creating courses, course cards and requests. Also let's please cover all path in separate tests.
- The case when a course has `course_rerun_object` and when doesn't has `course_rerun_object`
- The case when a user doesn't has any interest
- The case when there are no settings in a course.
- The case when there are no tags in a course.
- The case when form is `onboarding` and the case when form is not `onboarding`.

And let's please assert on case specific things, for example in case when form is `onboarding` we are adding `start_date`, `short description`, `interests` in `course` object, we should insert on these that these properties only exist when form is `onboarding`

Reply:
Can we please specify in the name of the test, the case we are testing in this test?

Reply:
what will be the name?